### ISSUE TITLE
feature request: allow float for process count

### ISSUE THREAD

**Issue by jjb** [sidekiq/sidekiq#6244]
In my environment, the number of cpus on a container is available in an environment variable, `CONTAINER_CPU_SHARE`.

I would like to pass this to `SIDEKIQ_COUNT`

problem is, it can be a "half cpu", e.g. `"2.5"`

Maybe sidekiq could accept this and round up, with something like `.to_f.ceil`?

I would vote for rounding up, to take full advantage of the half cpu. This is on the side of slightly higher memory use, but  my uninformed guess is most users who are passing in a value would be savvy enough to be okay with that.

```ruby
@count = Integer(env["SIDEKIQ_COUNT"].to_f.ceil || Etc.nprocessors)
```

**Comment by mperham** [comment]
```diff
-        @count = Integer(env["SIDEKIQ_COUNT"] || Etc.nprocessors)
+        @count = Float(env["SIDEKIQ_COUNT"] || Etc.nprocessors).round
```

This will round to the nearest integer: 2.4 -> 2 and 2.5 -> 3. wdyt?

**Comment by mperham** [comment]
Sidekiq Enterprise 7.2.3 now available.

**Comment by jjb** [comment]
FYI, on my container running on aptible:

```
> ENV['APTIBLE_CONTAINER_CPU_SHARE']
=> "0.25"
> Etc.nprocessors
=> 2 # this is wrong, it's the host
> Float(ENV['APTIBLE_CONTAINER_CPU_SHARE']).round
=> 0
> Float(ENV['APTIBLE_CONTAINER_CPU_SHARE']).ceil
=> 1
```

i think it's cgroups1

so, this is probably a reason to either use `ceil` or detect `0` and fall back to `1`. looks like `def start` doesn't do any other checks and will invoke `0.times`.

**Comment by mperham** [comment]
Will be fixed in 8.0.2.
